lib/metasploit/framework/common_engine.rb

```diff
@@ -33,12 +33,14 @@ module Metasploit::Framework::CommonEngine
     end
 
     config.root = Msf::Config::install_root
+    config.paths.add 'app/models', autoload: true
     config.paths.add 'app/concerns', autoload: true
     config.paths.add 'data/meterpreter', glob: '**/ext_*'
     config.paths.add 'modules'
 
     config.active_support.deprecation = :stderr
 
+    ActiveRecord.legacy_connection_handling = false
```

Comment:
This is a class level constant should we suppress this of check for `Rails 7` before setting?  Per Rails 7 legacy support is deprecated.

Reply:
Is there any additional context on this change? It looks like the console boots up without warnings for me without this being enabled 👀

Reply:
Rspec warns about the deprecation when run locally.
```
$ bundle exec rake spec
DEPRECATION WARNING: Using legacy connection handling is deprecated. Please set
`legacy_connection_handling` to `false` in your application.

The new connection handling does not support `connection_handlers`
getter and setter.

Read more about how to migrate at: https://guides.rubyonrails.org/active_record_multiple_databases.html#migrate-to-the-new-connection-handling
```

Reply:
Should we future proof this some?
```suggestion
      if ActiveRecord.respond_to?(:legacy_connection_handling)
        ActiveRecord.legacy_connection_handling = false
      end
```